### PR TITLE
new feature: filter tags and folder list

### PR DESCRIPTION
**Pull request by elfman**
## Description

for #2027
new feature: Filter tags and folder list
search

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

File: browser/main/SideNav/SideNav.styl [hunk 1]
```diff
@@ -18,7 +18,37 @@
...
 
+.search
+  position relative
+  flex 1
+  display flex
+  max-height 0
+  overflow hidden
+  transition max-height .4s
+  margin -5px 10px 0
+  .search-input
```
Comment (ZeroX-DG) [inline browser/main/SideNav/SideNav.styl]: Can you change the background of the input according to the theme? For example, in Monokai theme here is the background of the search input:
image

The background of other inputs:
image

File: browser/main/SideNav/SideNav.styl [hunk 2]
```diff
@@ -18,7 +18,37 @@
   display flex
   text-align center
 
+.extra-buttons
```
Comment (ZeroX-DG) [inline browser/main/SideNav/SideNav.styl]: The position doesn't feel right to me.

File: browser/main/SideNav/index.js [hunk 3]
```diff
@@ -438,9 +479,24 @@ class SideNav extends React.Component {
...
     const style = {}
     if (!isFolded) style.width = this.props.width
     const isTagActive = location.pathname.match(/tag/)
+
+    const navSearch = (
+      <div styleName='search' style={{maxHeight: showSearch ? '3em' : '0'}}>
+        <input
+          styleName='search-input'
+          type='text'
+          ref={dom => { this.searchInput = dom }}
```
Comment (ZeroX-DG) [inline browser/main/SideNav/index.js]: I don't see where this `ref` is used. Please remove it if you don't use it. :smiley:

**Comment by ZeroX-DG** [comment]
@elfman ping

**Comment by elfman** [comment]
@ZeroX-DG updated, please review again. Sorry, I'm too busy these days, and almost forgot this pr

**Comment by Flexo013** [comment]
Closing this as #2964 is merged and has the same function.

**Pull request closed without merging**